### PR TITLE
Implement the main reconcile

### PR DESCRIPTION
**Pull request by wangcanfengxs**
1.Make clear the HarborClusterConditionType
2.Implement the main reconcile

### CODE REVIEW

File: lcm/lcm.go [hunk 1]
```diff
@@ -24,26 +24,6 @@ type Controller interface {
 }
 
 type CRStatus struct {
-	Phase      Phase      `json:"phase"`
-	Properties Properties `json:"properties"`
+	Condition  v1.HarborClusterCondition `json:"condition"`
```
Comment (Ghostbaby) [inline lcm/lcm.go:27]: `HarborClusterConditionType` is only defined ready status, maybe we should add other states, e.p: Pending, Creating?

File: main.go [hunk 2]
```diff
@@ -65,9 +68,11 @@ func main() {
...
 
 	if err = (&controllers.HarborClusterReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("HarborCluster"),
-		Scheme: mgr.GetScheme(),
+		Client:        mgr.GetClient(),
+		Log:           ctrl.Log.WithName("controllers").WithName("HarborCluster"),
+		Scheme:        mgr.GetScheme(),
+		RequeueAfter:  requeueAfter,
+		ServiceGetter: &controllers.ServiceGetterImpl{},
```
Comment (Ghostbaby) [inline main.go:75]: Maybe we should add `Recorder`.
```suggestion
		ServiceGetter: &controllers.ServiceGetterImpl{},
		Recorder: mgr.GetEventRecorderFor("HarborCluster-Controller"),
```

File: api/v1/harborcluster_types.go [hunk 3]
```diff
@@ -344,18 +344,25 @@ type HarborClusterStatus struct {
...
+	// Ready means the HarborCluster is ready.
+	Ready HarborClusterConditionType = "Ready"
+	// CacheReady means the Cache is ready.
+	CacheReady HarborClusterConditionType = "CacheReady"
+	// DatabaseReady means the Database is ready.
+	DatabaseReady HarborClusterConditionType = "DatabaseReady"
+	// StorageReady means the Storage is ready.
+	StorageReady HarborClusterConditionType = "StorageReady"
+	// ServiceReady means the Service of Harbor is ready.
+	ServiceReady HarborClusterConditionType = "ServiceReady"
```
Comment (soulseen) [inline api/v1/harborcluster_types.go:363]: why not use the type of bool directly?

File: controllers/harborcluster_controller.go [hunk 4]
```diff
@@ -17,11 +17,9 @@ package controllers
...
-	"github.com/goharbor/harbor-cluster-operator/controllers/cache"
-	"github.com/goharbor/harbor-cluster-operator/controllers/database"
-	"github.com/goharbor/harbor-cluster-operator/controllers/harbor"
-	"github.com/goharbor/harbor-cluster-operator/controllers/storage"
 	"github.com/goharbor/harbor-cluster-operator/lcm"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"time"
 
 	"github.com/go-logr/logr"
```
Comment (soulseen) [inline controllers/harborcluster_controller.go:25]: how about using `klog`?

Comment (wangcanfengxs) [reply controllers/harborcluster_controller.go:25]: kb recommend to use it.

File: api/v1/harborcluster_types.go (hunk 3 above)
Comment (wangcanfengxs) [reply api/v1/harborcluster_types.go:363]: It just refer to k8s api-conventions.
`
  Type               FooConditionType   `json:"type" description:"type of Foo condition"`
  Status             ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`

  // +optional
  Reason             *string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
  // +optional
  Message            *string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`

  // +optional
  LastHeartbeatTime  *unversioned.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
  // +optional
  LastTransitionTime *unversioned.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`

`

File: lcm/lcm.go (hunk 1 above)
Comment (wangcanfengxs) [reply lcm/lcm.go:27]: Pending, Creating statuses are included in the Ready Status `False`. we can describe detail info in `Message` or  `Reason` field.

File: main.go (hunk 2 above)
Comment (wangcanfengxs) [reply main.go:75]: We haven't designed the event about the HarborClusterCR, we can add it next iteration.

**Pull request merged**